File: Leetcode/2089. Find Target Indices After Sorting Array/solution.py
class Solution:
    def targetIndices(self, nums: List[int], target: int) -> List[int]:
        smaller_than_target_count = 0
        target_count = 0
        
        for num in nums:
            if num < target:
                smaller_than_target_count += 1
            if num == target:
                target_count += 1
        
        return [i + smaller_than_target_count for i in range(target_count)]

File: Leetcode/2089. Find Target Indices After Sorting Array/solution.rs
impl Solution {
    pub fn target_indices(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut smaller_than_target_count = 0;
        let mut target_count = 0;

        for num in nums.iter() {
            if *num < target {
                smaller_than_target_count += 1;
            }
            if *num == target {
                target_count += 1;
            }
        }

        (smaller_than_target_count..smaller_than_target_count + target_count).collect()
    }
}
